nixos/modules/system/boot/systemd.nix

```diff
@@ -695,6 +697,11 @@ in
         minsize = "1M";
       };
     };
+
+    # make sure /run/next-system isn't garbage collected before the next boot
+    systemd.tmpfiles.rules = [ "L+ /nix/var/nix/gcroots/next-system - - - - /run/next-system" ];
+    # if /run/next-system exists, activate it before  rebooting
+    systemd.services.systemd-soft-reboot.serviceConfig.ExecStartPre = "-/run/next-system/activate";
```

Comment:
I am getting the following logs:

```
May 07 21:11:02 utm systemd[1]: Reached target System Shutdown.
May 07 21:11:02 utm systemd[1]: Reached target Late Shutdown Services.
May 07 21:11:02 utm systemd[1]: Starting Reboot System Userspace...
May 07 21:11:02 utm (activate)[6377]: systemd-soft-reboot.service: Failed to connect stdout to the journal socket, ignoring: Connection refused
May 07 21:11:02 utm systemd[1]: systemd-soft-reboot.service: Deactivated successfully.
May 07 21:11:02 utm systemd[1]: Finished Reboot System Userspace.
```


`systemd-soft-reboot.service` executes very late (`After=shutdown.target`) .  At this point running `activate` probably doesn't work anymore. Also see the

```
May 07 21:11:02 utm (activate)[6377]: systemd-soft-reboot.service: Failed to connect stdout to the journal socket, ignoring: Connection refused
```


so this probably isn't working as expected